Return user info when making a social post (Fixes #48)

kyl191 opened the pull request:
* Shifts some stuff around to create defaults in one place.
* Changes existing code to not use defaults.
* Adds the user name + profile pic URL to the data that's returned when the user makes a post

(cc @ericljpemberton)

kyl191 committed:
Consolidate defaults into the uid -> name mapping function instead of duplicating

kyl191 committed:
#48: pass the caller name & icon back to the app

TheKevJames commented on api/models/social.py:
```diff
@@ -49,9 +51,12 @@ def map_id_to_names(uids):
-        mapping[attrs['id']] = {'name': ' '.join((attrs.get('first_name', ''),
-                                                  attrs.get('last_name', ''))),
-                                'poster_icon': attrs.get('photo', '')}
+        first_name = attrs.get('first_name', 'Random')
```
Setting a default username should be a frontend thing, not a backend thing. Send `'name': ''` and have them default to whatever they want.

kyl191 committed:
Provide empty string defaults for first + last name.
Also, strip the created name to send an empty string instead of a string with a single space.

kyl191 committed:
I need a pre-commit pep8 check
